dissect/cstruct/cstruct.py

```diff
@@ -263,6 +290,93 @@ def resolve(self, name: str) -> BaseType:
 
         raise ResolveError(f"Recursion limit exceeded while resolving type {name}")
 
+    def _make_type(
+        self, name: str, bases: Iterator[object], size: int, *, alignment: int = None, attrs: dict[str, Any] = None
+    ) -> type[BaseType]:
+        attrs = attrs or {}
+        attrs.update(
+            {
+                "cs": self,
+                "size": size,
+                "dynamic": size is None,
+                "alignment": alignment or size,
+            }
+        )
+        return types.new_class(name, bases, {}, lambda ns: ns.update(attrs))
+        # return type(name, bases, attrs)
+
+    def _make_array(self, type_: MetaType, num_entries: Optional[UnionHint[int, Expression]]) -> type[Array]:
+        null_terminated = num_entries is None
+        dynamic = isinstance(num_entries, Expression)
+        size = None if null_terminated or dynamic else num_entries * type_.size
+        name = f"{type_.__name__}[]" if null_terminated else f"{type_.__name__}[{num_entries}]"
+
+        bases = (Array,)
+        if issubclass(type_, Char):
+            bases = (CharArray,)
+        elif issubclass(type_, Wchar):
+            bases = (WcharArray,)
+
+        attrs = {
+            "type": type_,
+            "num_entries": num_entries,
+            "null_terminated": null_terminated,
+            "dynamic": dynamic,
+        }
+
+        return self._make_type(name, bases, size, alignment=type_.alignment, attrs=attrs)
+
+    def _make_int_type(self, name: str, size: int, signed: bool, *, alignment: int = None) -> type[Int]:
+        return self._make_type(name, (Int,), size, attrs={"signed": signed, "alignment": alignment})
```

Comment:
```suggestion
        return self._make_type(name, (Int,), size, alignment=alignment, attrs={"signed": signed})
```

Otherwise it will get overwritten with `size`.

Reply:
Indeed, nice catch!